pallets/parachain-system/src/validate_block/implementation.rs

```diff
@@ -65,11 +65,17 @@ pub fn validate_block<B: BlockT, E: ExecuteBlock<B>, PSC: crate::Config>(
 		"Invalid parent hash",
 	);
 
-	let db = storage_proof.into_memory_db();
-	let root = parent_head.state_root().clone();
-	if !HashDB::<HashFor<B>, _>::contains(&db, &root, EMPTY_PREFIX) {
-		panic!("Witness data does not contain given storage root.");
-	}
+	// Uncompress
+	let mut db = MemoryDB::default();
+	let root = match sp_trie::decode_compact::<sp_trie::Layout<HashFor<B>>, _, _>(
+		&mut db,
+		storage_proof.iter_compact_encoded_nodes(),
+		Some(parent_head.state_root()),
+	) {
```

Comment:
Ahh and didn't you added a method to the compactproof for uncompressing it?

Reply:
We would have to rebuild the db afterward, here we directly got the db with the right content.

Reply:
Ahh okay